sqlair.go

```diff
@@ -92,15 +97,98 @@ type Iterator struct {
 	started bool
 }
 
+// A Least Recently Used Cache using a doubly linked list.
+type lruCache struct {
+	ll   *list.List
+	c    map[index]*list.Element
+	size int
+}
+
+// The cache key.
+type index struct {
+	tx *TX
```

Comment:
The `TX` will only be in the cache if it hasn't been committed or rolled back. When they do do either of those operations it is removed. I don't think that it would keep the db connection alive even if the `TX` was not removed as long as it had been committed or rolled back the connection is closed. If there is a situation that arises where the cache api is misused like this then the tests would fail.

I'm not too worried about this in general since its not a public api and its not going to be widely used within the program.

Reply:
This makes sense about it not being a public API, it'll be easier to police correct usage of this as it's internal.

I see in the definition of `TX` that there's (indirectly) a `*sql.DB` which would be kept alive which made me suspicious